### PR TITLE
Do not use array like access of Model

### LINKED ISSUE #1195: Do not use array access to access Model data
related with PR https://github.com/atk4/data/pull/591

must be fixed in any case asap

should be also tested in CI

### PR DESCRIPTION
**Pull request by mvorisek**
fix #1195

### CODE REVIEW

**Comment by mvorisek** [comment]
Please review and merge, no function change, no BC break.

**Comment by mvorisek** [comment]
@ibelar https://codecov.io/gh/atk4/ui/compare/0ada1e5cae397573b11f703c6cafdf865007296b...0686e4336cc317afe49edeeb94127701e428f3f0/src/src/TableColumn/Multiformat.php this PR seems that it changed this, can you look at it?

File: demos/database.php [hunk 1]
```diff
@@ -129,12 +129,12 @@ public function init(): void
         $this->onHook('afterLoad', function ($m) {
```
Comment (DarkSide666) [inline demos/database.php]: you can add atk4/data/Model hint there

File: demos/form/form2.php [hunk 2]
```diff
@@ -39,29 +39,29 @@
...
+        $errors[] = $form->error('first_name', 'too short, ' . $form->model->get('first_name'));
     }
-    if (mb_strlen($f->model['last_name']) < 5) {
-        $errors[] = $f->error('last_name', 'too short');
+    if (mb_strlen($form->model->get('last_name')) < 5) {
+        $errors[] = $form->error('last_name', 'too short');
     }
-    if (isset($f->model->dirty['iso'])) { // restrict to change iso field value
-        $errors[] = $f->error('iso', 'Field value should not be changed');
+    if (isset($form->model->dirty['iso'])) { // restrict to change iso field value
```
Comment (DarkSide666) [inline demos/form/form2.php]: It's a bit worse for performance, but I still would better use `isDirty()` method there and not access models property directly.

File: demos/input/lookup.php [hunk 3]
```diff
@@ -32,8 +32,8 @@
     'search' => ['name', 'iso', 'iso3'],
 ]);
 
-$form->onSubmit(function ($f) use ($db) {
-    $str = $f->model->ref('country1')['name'] . ' ' . $f->model->ref('country2')['name'] . ' ' . (new Country($db))->tryLoad($f->model['country3'])->get('name');
+$form->onSubmit(function (atk4\ui\Form $form) use ($db) {
+    $str = $form->model->ref('country1')['name'] . ' ' . $form->model->ref('country2')['name'] . ' ' . (new Country($db))->tryLoad($form->model->get('country3'))->get('name');
```
Comment (DarkSide666) [inline demos/input/lookup.php]: should be `ref(...)->get('name')` here

File: src/FormField/MultiLine.php [hunk 4]
```diff
@@ -932,19 +932,19 @@ private function getDummyExpression($exprField, $model)
...
-    private function getValueForExpression($exprField, $fieldName, $model)
+    private function getValueForExpression($exprField, $fieldName, Model $m)
     {
         switch ($exprField->type) {
             case 'money':
             case 'integer':
             case 'float':
                 // Value is 0 or the field value.
-                $value = $model[$fieldName] ? $model[$fieldName] : 0;
+                $value = $m->get($fieldName) !== '' ? $m->get($fieldName) : 0;
```
Comment (DarkSide666) [inline src/FormField/MultiLine.php]: what if get() returns `null`? Is that possible? This looks like actual change in code

File: src/FormField/MultiLine.php [hunk 5]
```diff
@@ -932,19 +932,19 @@ private function getDummyExpression($exprField, $model)
...
 
                 break;
             default:
                 // Value is "" or field value enclosed in bracket: "value"
-                $value = $model[$fieldName] ? '"' . $model[$fieldName] . '"' : '""';
+                $value = $m->get($fieldName) !== '' ? '"' . $m->get($fieldName) . '"' : '""';
```
Comment (DarkSide666) [inline src/FormField/MultiLine.php]: what if get() returns `null`? Is that possible? This looks like actual change in code

File: src/TableColumn/Tooltip.php [hunk 6]
```diff
@@ -64,12 +65,12 @@ public function getDataCellHTML(Field $f = null, $extra_tags = [])
...
 
-    public function getHTMLTags($row, $field)
+    public function getHTMLTags(Model $row, $field)
     {
         // @TODO remove popup tooltip when null
-        $tooltip = $row->data[$this->tooltip_field];
+        $tooltip = $row->get($this->tooltip_field);
 
-        if ($tooltip === null || empty($tooltip)) {
+        if ($tooltip === null || $tooltip === '') {
```
Comment (DarkSide666) [inline src/TableColumn/Tooltip.php:73]: what if tooltip is `null`? actual BC change

**Review by DarkSide666 — CHANGES_REQUESTED** [review]
See comments

File: src/TableColumn/Tooltip.php (hunk 6 above)
Comment (mvorisek) [reply src/TableColumn/Tooltip.php:73]: Looking, looking, still think `null` is checked

File: src/FormField/MultiLine.php (hunk 4 above)
Comment (mvorisek) [reply src/FormField/MultiLine.php]: Feel free to commit a fix on top of this PR, quite hard to fix everything, 81 changed files...

File: demos/input/lookup.php (hunk 3 above)
Comment (mvorisek) [reply demos/input/lookup.php]: Yes, commit a fix, another example of mixed usage on the same line :D

File: demos/form/form2.php (hunk 2 above)
Comment (mvorisek) [reply demos/form/form2.php]: Metoo!

File: src/TableColumn/Tooltip.php (hunk 6 above)
Comment (DarkSide666) [reply src/TableColumn/Tooltip.php:73]: oh, silly me :D :D :)

File: demos/database.php (hunk 1 above)
Comment (mvorisek) [reply demos/database.php]: Skipping this in this PR, much more to add - I fexed completedy only some specific hook names.

File: demos/input/lookup.php (hunk 3 above)
Comment (DarkSide666) [reply demos/input/lookup.php]: fixed

**Review by DarkSide666 — APPROVED** [review]

**Review by DarkSide666 — APPROVED** [review]

**Comment by mvorisek** [comment]
@DarkSide666 merge once you are done with merging :)

**Pull request merged**